### PR TITLE
Displayer hl_chars and TelescopeResults hlgroups

### PR DESCRIPTION
**Pull request by Conni2461**
~~Will remove the filename fix asap~~

CC @sunjon

### CODE REVIEW

File: lua/telescope/make_entry.lua [hunk 1]
```diff
@@ -294,15 +294,25 @@ function make_entry.gen_from_symbols(opts)
   opts = opts or {}
   opts.tail_path = get_default(opts.tail_path, true)
 
-    local displayer = entry_display.create {
+  local displayer = entry_display.create {
     separator = "",
+    hl_chars = { ['['] = 'TelescopeBorder', [']'] = 'TelescopeBorder' },
```
Comment (Conni2461) [inline lua/telescope/make_entry.lua:299]: @sunjon Thats the new interface. Any comments?

File: lua/telescope/make_entry.lua [hunk 2]
```diff
@@ -294,15 +294,25 @@ function make_entry.gen_from_symbols(opts)
...
-    },
+    }
+  }
+
+  local default_type_highlight = {
+    ["Class"]    = "TelescopeResultsClass",
+    ["Constant"] = "TelescopeResultsConstant",
+    ["Field"]    = "TelescopeResultsField",
+    ["Function"] = "TelescopeResultsFunction",
+    ["Method"]   = "TelescopeResultsMethod",
```
Comment (Conni2461) [inline lua/telescope/make_entry.lua:312]: Added `Method` as well

File: lua/telescope/make_entry.lua [hunk 3]
```diff
@@ -538,20 +536,17 @@ end
 
 function make_entry.gen_from_registers(_)
   local displayer = entry_display.create {
-    separator = "",
+    separator = " ",
```
Comment (Conni2461) [inline lua/telescope/make_entry.lua:539]: I believe we want a space here now right?

File: lua/telescope/make_entry.lua [hunk 4]
```diff
@@ -294,15 +294,25 @@ function make_entry.gen_from_symbols(opts)
...
       { width = 6 },
       { width = 40 },
-      { width = 1 },
       { remaining = true },
-      { width = 1 },
-    },
+    }
+  }
+
+  local default_type_highlight = {
```
Comment (Conni2461) [inline lua/telescope/make_entry.lua:307]: Btw i moved that table out. We might wanna have that outside of this scope as well so it won't get redefined that often.

File: lua/telescope/make_entry.lua (hunk 1 above)
Comment (sunjon) [reply lua/telescope/make_entry.lua:299]: Seems reasonable. I've used it in the new LSP layout for inline fileinfo.

Comment (Conni2461) [reply lua/telescope/make_entry.lua]: Saw it :)

**Pull request merged**